STYLE.md

```diff
@@ -52,6 +52,8 @@
   exceptions such as `main()` functions. When code cannot be placed inside the
   Envoy namespace there should be a comment of the form `// NOLINT(namespace-envoy)` at
   the top of the file.
+* If a public method is only intended to be called from test code then it should
+  have a name that ends in ForTest() such as aMethodForTest().
```

Comment:
I didn't call it out here explicitly but my understanding is that in Envoy style this is preferable to doing something with `friend`-ing the test code. Thoughts on that?